src/components/vote/VoteButton.tsx

```diff
@@ -1,24 +1,51 @@
 import { useTranslation } from 'react-i18next';
+import { Contracts } from '@ardenthq/sdk-profiles';
+import { useMemo } from 'react';
 import { Button } from '@/shared/components';
 import { Footer } from '@/shared/components/layout/Footer';
 
 export const VoteButton = ({
-    disabled,
+    delegateAddress,
+    fee: _,
+    votes,
     onClick,
-    type = 'vote',
+    isValid: _2,
 }: {
-    disabled: boolean;
+    delegateAddress?: string;
+    fee: string;
+    votes: Contracts.VoteRegistryItem[];
     onClick: () => void;
-    type?: 'vote' | 'unvote' | 'swap_vote';
+    isValid: boolean;
 }) => {
+    // @TODO: disable if empty fee and form is not valid once fee is added
+    // const disabled = !isValid || delegateAddress === undefined || fee === '';
+    const disabled = delegateAddress === undefined;
+
+    const isVoted = useMemo(() => {
+        if (delegateAddress === undefined) {
+            return false;
+        }
+
+        return votes.some((vote) => vote.wallet?.address() === delegateAddress);
+    }, [votes, delegateAddress]);
+
+    const isSwapping = useMemo(() => votes.length > 0 && !isVoted, [votes, isVoted]);
```

Comment:
A suggestion: you could get rid of `useMemo` as these are rather simple calculations..